### PR TITLE
BREAKING: remove support for JSON imports

### PR DESCRIPTION
**Pull request by bartlomieju**
This commit removes support for importing JSON files as modules.

This change is dictated by security; browsers rolled back the support.

### CODE REVIEW

File: cli/global_state.rs [hunk 1]
```diff
@@ -118,8 +115,9 @@ impl GlobalState {
     let compile_lock = self.compile_lock.lock().await;
 
     let compiled_module = match out.media_type {
-      msg::MediaType::Unknown => state1.js_compiler.compile(out).await,
-      msg::MediaType::Json => state1.json_compiler.compile(&out).await,
+      msg::MediaType::Json | msg::MediaType::Unknown => {
```
Comment (bartlomieju) [inline cli/global_state.rs:118]: I think `MediaType::Json` could now be removed; it'd be handled as `MediaType::Unknown`

**Comment by bartlomieju** [comment]
CC @kitsonk

**Comment by nayeemrmn** [comment]
Closes 3401.

**Comment by ry** [comment]
There may be special handling of json imports in the bundler code base.

File: cli/tests/integration_tests.rs [hunk 2]
```diff
@@ -395,38 +395,6 @@ fn bundle_single_module() {
...
-  let json_modules = util::root_path().join("cli/tests/020_json_modules.ts");
-  assert!(json_modules.is_file());
-  let t = TempDir::new().expect("tempdir fail");
-  let bundle = t.path().join("020_json_modules.bundle.js");
-  let mut deno = util::deno_cmd()
-    .current_dir(util::root_path())
-    .arg("bundle")
-    .arg(json_modules)
-    .arg(&bundle)
-    .spawn()
```
Comment (bartlomieju) [inline cli/tests/integration_tests.rs:409]: @ry thanks for pointing this out; I guess there should be permission check involved in this case like a regular disk read?

**Comment by kitsonk** [comment]
LGTM

**Pull request merged**